generate_qt_mappings.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+"""
+This script generated the Qt mapping file according to given Qt include
+directory
+
+Example usage :
+
+   $ ./generate_qt_mappings.py /usr/include/x86_64-linux-gnu/qt5 qt5_11.imp
+
+"""
+
+import argparse
+import glob
+import os
+import re
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("qt_include_dir", help="Qt include directoy")
+    parser.add_argument("output_file", help="Generated output mapping file")
+    parser.add_argument("--warnings", action="store_true", help="Display possible warning messages")
+    args = parser.parse_args()
+
+    headers = glob.glob(os.path.join(args.qt_include_dir, '**/*[!\\.h]'))
+    symbols_map = []
+    includes_map = []
+
+    class2includes = {}
+    include2classes = {}
+    for header in headers:
+
+        if os.path.isdir(header):
+            continue
+
+        class_name = os.path.basename(header)
+        module_name = os.path.basename(os.path.dirname(header))
+
+        symbols_map += [f'{{ symbol: [ "{class_name}", "private", '
+                        f'"<{class_name}>", "public" ] }}']
+
+        content = open(header, 'r').read()
+        includes = re.findall(r'#include "(.*)\.h"', content)
+        if len(includes) != 1:
+            continue
+
+        include = includes[0]
+        class2includes[class_name] = (class2includes.get(class_name, [])
```

Comment:
This is what `setdefault` is for:

    class2includes.setdefault(class_name, []).append(header)